### ISSUE TITLE
Encryption consistency issues with untrusted devices

### ISSUE THREAD

**Issue by ginkel** [Catfriend1/syncthing-android#804]
### Description of the issue

I have set up three Syncthing instances: One running on my laptop (v1.16.1 on ArchLinux), one running on my Android smartphone (Syncthing-Fork v1.16.0.3 from Google Play) and another one on a server based on Syncthing's Docker image.

The instance running on a server is marked as untrusted and I have set up passwords for all folders shared with it (for Syncthing-Fork I had to go through the Web-UI to do this). All folders are set up as "Receive Encrypted" on the server instance.

This worked successfully for a few hours, but recently I started getting the following message on the server instance:

> Failure checking encryption consistency with device pixel5.DOMAIN for folder "Keepass" (keepass): folder is configured to be encrypted but not announced thus

### Reproduction Steps

This might turn out to be difficult due to the fact that everything seems to work fine for a while until the encryption property is suddenly dropped.

### Version Information
- App Version: 1.16.0.3
- Syncthing Version: v1.16.0 (app), v1.16.1 (Linux)
- Android Version: Android 11
- Device manufacturer: Google
- Device model: Pixel 5

### Device platform info

```sh
[ro.product.board]: [redfin]
[ro.product.brand]: [google]
[ro.product.build.date]: [Fri Apr  2 19:16:07 UTC 2021]
[ro.product.build.date.utc]: [1617390967]
[ro.product.build.fingerprint]: [google/redfin/redfin:11/RQ2A.210505.003/7255357:user/release-keys]
[ro.product.build.id]: [RQ2A.210505.003]
[ro.product.build.tags]: [release-keys]
[ro.product.build.type]: [user]
[ro.product.build.version.incremental]: [7255357]
[ro.product.build.version.release]: [11]
[ro.product.build.version.release_or_codename]: [11]
[ro.product.build.version.sdk]: [30]
[ro.product.cpu.abi]: [arm64-v8a]
[ro.product.cpu.abilist]: [arm64-v8a,armeabi-v7a,armeabi]
[ro.product.cpu.abilist32]: [armeabi-v7a,armeabi]
[ro.product.cpu.abilist64]: [arm64-v8a]
[ro.product.device]: [redfin]
[ro.product.first_api_level]: [30]
[ro.product.locale]: [en-US]
[ro.product.manufacturer]: [Google]
[ro.product.model]: [Pixel 5]
[ro.product.name]: [redfin]
[ro.product.odm.brand]: [google]
[ro.product.odm.device]: [redfin]
[ro.product.odm.manufacturer]: [Google]
[ro.product.odm.model]: [Pixel 5]
[ro.product.odm.name]: [redfin]
[ro.product.product.brand]: [google]
[ro.product.product.device]: [redfin]
[ro.product.product.manufacturer]: [Google]
[ro.product.product.model]: [Pixel 5]
[ro.product.product.name]: [redfin]
[ro.product.system.brand]: [google]
[ro.product.system.device]: [generic]
[ro.product.system.manufacturer]: [Google]
[ro.product.system.model]: [mainline]
[ro.product.system.name]: [mainline]
[ro.product.system_ext.brand]: [google]
[ro.product.system_ext.device]: [redfin]
[ro.product.system_ext.manufacturer]: [Google]
[ro.product.system_ext.model]: [Pixel 5]
[ro.product.system_ext.name]: [redfin]
[ro.product.vendor.brand]: [google]
[ro.product.vendor.device]: [redfin]
[ro.product.vendor.manufacturer]: [Google]
[ro.product.vendor.model]: [Pixel 5]
[ro.product.vendor.name]: [redfin]
[ro.product.vndk.version]: [30]
```

### Android Log

```sh

```

**Comment by Catfriend1** [comment]
Hi,

encryption on the wrapper's UI is not ready yet. I didn't have the time to finish my PR and test. The backend is roughly prepared so setting up through the web ui was a good idea of you already.

To narrow it down: 
- does restarting the app cause the problem ? (main menu - restart)
- does editing a folder of your choice from the wrapper to e.g. pause/unpause it cause the problem?
- if you have skills and time you could try to setup a test folder pair and immediately after that use "adb pull path/to/config.xml" to save it on your computer. then do that same thing again and again multiple times after changing settings on folders via the wrapper ui, restarting the app ... and finally after your problem became visible. I  suspect some config model in the app causing this. Compare those captures config.xml's and see in text editor if anything got lost you didn't have put your hands on.

**Comment by Catfriend1** [comment]
To be reopened if more info is available.
I think a good start to debug this is pulling the config.xml file at multiple steps off the phone and publish it here (sensitive information redacted/replaced by devA, devB, folderA, folderB, ...).

```
adb shell "su root cat /data/data/com.github.catfriend1.syncthingandroid/files/config.xml" > "config.xml"
```

and for the "debug" version:
```
adb shell "su root cat /data/data/com.github.catfriend1.syncthingandroid.debug/files/config.xml" > "config.xml"
```

**Comment by nunesgh** [comment]
I have experienced something similar to this issue, but I cannot provide additional information.

It seems to be similar to this issue, though: https://github.com/syncthing/syncthing-android/issues/1673.

**Comment by Catfriend1** [comment]
To be reopened if more info is available.

P.S. encryption support is not completed yet, see https://github.com/Catfriend1/syncthing-android/pull/725

**Comment by nunesgh** [comment]
Hi @Catfriend1!

I can confirm that changes to a folder using the Android UI instead of the Web UI is what causes all encrypted folders to have their passwords erased in Syncthing-Fork.

I have a configuration similar to the following one. Dotted lines are E2EE. All folders in safe devices have two E2EE shares with unsafe devices and one normal share with a safe device. Unsafe devices also share their encrypted folders between them.

![image](https://user-images.githubusercontent.com/16168765/132955253-6eca9e1e-2a73-4667-a9ce-749eef75afc4.png)

I updated `Folder 2` in Syncthing-Fork (ST-F) using the Android UI to add an Ignore Pattern to it. As a result, passwords for `Folder 1` and `Folder 2` were erased from ST-F and I had to re-add the passwords through the Web UI to restore E2EE sync with the unsafe devices.

Otherwise, the setup has been working as expected if no changes are made in ST-F through the Android UI.

Thank you for your attention.
